magicctapipe/scripts/lst1_magic/lst1_magic_train_rfs.py

```diff
@@ -35,410 +39,306 @@
+    logger.info(
+        f"\nFeatures:\n{features}" f"\n\nUse unsigned features: {use_unsigned_features}"
     )
 
-    # Train the regressors per telescope combination:
+    energy_regressor = EnergyRegressor(rf_settings, features, use_unsigned_features)
+
+    # Train the RFs per telescope combination type
     Path(output_dir).mkdir(exist_ok=True, parents=True)
 
+    output_files = []
+
     for tel_combo in data_train.keys():
 
-        logger.info(f'\nTraining energy regressors for "{tel_combo}" events...')
+        logger.info(f"\nEnergy regressors for the '{tel_combo}' type:")
         energy_regressor.fit(data_train[tel_combo])
 
-        logger.info("\nFeature importance:")
-        check_feature_importance(energy_regressor)
+        # Check the feature importance
+        for tel_id, telescope_rf in energy_regressor.telescope_rfs.items():
 
+            logger.info(f"\n{TEL_NAMES[tel_id]} feature importance:")
+            importances = telescope_rf.feature_importances_
+
+            for feature, importance in zip(features, importances):
+                logger.info(f"\t{feature}: {importance.round(5)}")
+
+        # Save the trained RFs to an output file
         if use_unsigned_features:
             output_file = f"{output_dir}/energy_regressors_{tel_combo}_unsigned.joblib"
         else:
             output_file = f"{output_dir}/energy_regressors_{tel_combo}.joblib"
 
         energy_regressor.save(output_file)
-        logger.info(f"\nOutput file:\n{output_file}")
+        output_files.append(output_file)
 
+    logger.info("\nOutput file(s):")
+    for output_file in output_files:
+        logger.info(output_file)
 
-def train_direction_regressor(
-    input_file,
-    output_dir,
-    config,
-    use_unsigned_features=False,
-):
+
+def train_disp_regressor(input_dir, output_dir, config, use_unsigned_features=False):
     """
-    Trains direction regressors with input gamma MC DL1-stereo data.
+    Trains DISP regressors with gamma MC DL1-stereo events.
 
     Parameters
     ----------
-    input_file: str
-        Path to an input gamma MC DL1-stereo data file
+    input_dir: str
+        Path to a directory where input gamma MC data files are found
     output_dir: str
         Path to a directory where to save trained RFs
     config: dict
         Configuration for the LST-1 + MAGIC analysis
     use_unsigned_features: bool
-        If True, it uses unsigned features for training RFs
+        If `True`, it uses unsigned features for training RFs
     """
 
-    config_rf = config["direction_regressor"]
+    config_rf = config["disp_regressor"]
 
-    # Load the input file:
-    logger.info(f"Input file:\n{input_file}")
+    # Load the input files
+    logger.info(f"\nInput directory:\n{input_dir}")
 
-    data_train = load_train_data_file(
-        input_file,
-        config_rf["gamma_offaxis"]["min"],
-        config_rf["gamma_offaxis"]["max"],
-    )
+    offaxis_min = config_rf["gamma_offaxis"]["min"]
```

Comment:
this part of getting offaxis min/max and converting to Quantity is used 3 times, maybe extract it into a function?

Reply:
Thanks for the comment, I changed the code to do so.